src/ray/core_worker/core_worker.cc

```diff
@@ -1764,6 +1766,13 @@ void CoreWorker::HandleWaitForRefRemoved(const rpc::WaitForRefRemovedRequest &re
 void CoreWorker::HandleCancelTask(const rpc::CancelTaskRequest &request,
                                   rpc::CancelTaskReply *reply,
                                   rpc::SendReplyCallback send_reply_callback) {
+  // Handle remote cancelation requests.
+  if (!request.remote_object_id().empty()) {
```

Comment:
What if the remote object ID is empty? I'm a bit confused by this logic, why wouldn't we call `CancelTask` in all cases?

Reply:
Oh, I see, we're basically just forwarding the request in this case. I think overloading the RPC message is probably fine, but can you modify the comment above to make it more obvious what's happening? Left a suggestion

Reply:
That looks great! Thanks for that!